docs/design/CONFIG_API.md

```diff
@@ -0,0 +1,249 @@
+Configuration API Design
+===================
+
+## Motivation
+[Issue 51][issue_51] specifies the need to design an API for configuring Envoy Gateway at runtime. This configuration is
+referred to as the "static configuration" in the Envoy Gateway [design doc][design_doc].
+
+## Goals
+* Define an __initial__ API for configuring Envoy Gateway at runtime, independent of the runtime platform, e.g.
+  Kubernetes.
+
+## Non-Goals
+* Implementation of the Envoy Gateway runtime configuration API.
+* Define the `status` subresource of the runtime configuration API.
+* Define a __complete__ API for configuring Envoy Gateway at runtime. As stated in the [Goals](#goals), this document
+  defines the initial runtime configuration API for Envoy Gateway.
+* Define an API for deploying/provisioning/operating Envoy Gateway.
+* Define an __initial__ provider-independent API for configuring the managed Envoy proxy infrastructure.
+* Specify tooling for managing the API, e.g. generate protos, CRDs, controller RBAC, etc.
+
+## API Definition
+`EnvoyGatewayConfigSpec` defines the runtime configuration of Envoy Gateway.
```

Comment:
```suggestion
`EnvoyGatewayConfigSpec` defines the static configuration of Envoy Gateway.
```

Reply:
ack. Commit 9efda09 resolves this comment.